test/powershell/engine/Module/NewModuleManifest.Tests.ps1

```diff
@@ -116,4 +116,68 @@ Describe "New-ModuleManifest tests" -tags "CI" {
 
         { New-ModuleManifest -Path $manifestPath -ProjectUri $testUri -LicenseUri $testUri -IconUri $testUri } | Should -Throw -ErrorId "System.InvalidOperationException,Microsoft.PowerShell.Commands.NewModuleManifestCommand"
     }
+
+    It "New-ModuleManifest works with assembly architecture: <moduleArch>" -TestCases @(
+        # All values from [System.Reflection.ProcessorArchitecture]
+        @{ moduleArch = "None" },
+        @{ moduleArch = "MSIL" },
+        @{ moduleArch = "X86" },
+        @{ moduleArch = "IA64" },
+        @{ moduleArch = "Amd64" },
+        @{ moduleArch = "Arm" }
+    ) {
+        param($moduleArch)
+
+        $roslynArch = [System.Runtime.InteropServices.RuntimeInformation]::ProcessArchitecture
+
+        switch ($moduleArch)
+        {
+            "None"  { Set-ItResult -Skipped -Because "the test assembly architecture can not be tested"; return }
+            "MSIL"  { Set-ItResult -Skipped -Because "the test assembly architecture can not be tested"; return }
+            "IA64"  { $roslynArch = 'Itanium' }
+        }
+
+        $arch = [int].Assembly.GetName().ProcessorArchitecture
+
+        # Skip tests if the module architecture does not match the platform architecture
+        # but X86 works on Amd64/X64 and Arm works on Arm64.
+        if ($moduleArch -ne $arch -and -not ($moduleArch -eq "X86" -and $arch -eq "Amd64") -and -not ($moduleArch -eq "Arm" -and $arch -eq "Arm64"))
+        {
+            Set-ItResult -Skipped -Because "the $moduleArch assembly architecture is not supported on the $arch platform"
+            return
+        }
+
+        $a=@"
+    using System;
+    using System.Management.Automation;
+
+    namespace Test.Manifest {
+
+        [Cmdlet(VerbsCommon.Get, "TP")]
+        public class TPCommand0 : PSCmdlet
+        {
+            protected override void EndProcessing()
+            {
+                WriteObject("$arch");
+            }
+        }
+    }
+"@
+        # We can not unload the module assembly and so we can not use Pester TestDrive without cleanup error reporting
+        $testFolder = Join-Path $([System.IO.Path]::GetTempPath()) $([System.IO.Path]::GetRandomFileName())
+        New-Item -Type Directory -Path $testFolder -Force > $null
+
+        $assemblyPath = Join-Path $testFolder "TP_$arch.dll"
+        $modulePath = Join-Path $testFolder "TP_$arch.psd1"
+        Add-Type -TypeDefinition $a -CompilerOptions "/platform:$roslynArch" -OutputAssembly $assemblyPath
+        New-ModuleManifest -Path $modulePath -NestedModules "TP_$arch.dll" -RequiredAssemblies "TP_$arch.dll" -ProcessorArchitecture $arch -CmdletsToExport "Get-TP"
+        Import-Module $modulePath
+
+
+        Get-TP -ErrorAction SilentlyContinue | Should -BeExactly "$arch"
+
+        Remove-Module "TP_$arch"
+        Remove-Item -LiteralPath $testFolder -Recurse -Force -ErrorAction SilentlyContinue > $null
```

Comment:
Should have in `finally` to ensure this cleans up if test fails for some reason

Reply:
Done.